src/number-value.js

```diff
@@ -23,7 +23,13 @@
   }
   internal.inherit(NumberValue, internal.StyleValue);
 
+  NumberValue.parse = function(value) {
+    if (/^\s*[-+]?(\d*\.)?\d+(e[-+]?\d+)?\s*$/.test(value)) {
```

Comment:
Lets move this regex out into a parser file like in my pull request. There will be lots of other properties that will need to parse numbers as part of their parsing, and it makes sense to share this regex with them.